webapp/graphite/util.py

```diff
@@ -57,6 +57,8 @@ def epoch(dt):
   """
   Returns the epoch timestamp of a timezone-aware datetime object.
   """
+  if not dt.tzinfo:
+    return calendar.timegm(make_aware(dt, pytz.timezone(settings.TIME_ZONE)).astimezone(pytz.utc).timetuple())
```

Comment:
IIRC this was done on purpose: an epoch is a fixed point in time, whereas a naive datetime can have multiple interpretations. I'm not sure using settings.TIME_ZONE as default fallback is always correct.

The absence of fallback is a way to raise issues when naive datetime are passed around. The fix is to make these datetimes aware (as you did 👍), but I'd keep epoch() fail rather than being at risk of being off by X hours.

Reply:
I understand what you're saying, but at the same time a hard failure is a pretty nasty thing.  I think that using the configured timezone is a pretty reasonable approach, and if we're worried about it then I can easily add a "warning non-timezone aware datetime passed to epoch" log entry there.